### ISSUE TITLE
Problem with routing and swagger ui

### ISSUE THREAD

**Issue by fdundjer** [nestjs/swagger#1976]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Hello,

With new version of swagger package, Swagger UI stops working.
I've tested latest version 6.0.3

When opening Swagger UI page, I get 404 response on resource requests.
It fails to load:
- swagger-ui.css
- swagger-ui.bundle.js
- swagger-ui-standalone-preset.js
- swagger-ui-init.js

It looks like swagger is looking for resources on wrong location.

 On previous versions of package swagger-ui.css was loaded from:
- https://dev.address.com/api/service/swagger/swagger-ui.css" 

On new version of package, swagger-ui.css is loaded from:
- https://dev.address.com/swagger/swagger-ui.css" 

Service in both cases is running on:
https://dev.address.com/api/service/

We are using istio api gateway to configure our routes.
Configuration looks similar to this:

```
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: address.com
spec:
  hosts:
    - "*.address.com"
  gateways:
    - api-gateway
  http:
    - match:
        - uri:
            prefix: /api/service
        - uri:
            exact: /api/service
      rewrite:
        uri: " "
      route:
        - destination:
            host: service-cluster-ip-service
            port:
              number: 80
---
```

Similar issue: https://github.com/nestjs/swagger/issues/1965

### Minimum reproduction code

Our swagger setup:
https://gist.github.com/fdundjer/c09753d162da16602c1d16e5506670e7
Did we misconfigure something? 

### Expected behavior

Swagger UI page works as with previous package versions.

### Package version

6.0.3

### NestJS version

9.0.3

### Node.js version

v16.13.1

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by kamilmysliwiec** [comment]
cc @flamewow

**Comment by kamilmysliwiec** [comment]
Fixed in 6.0.4
